http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ProxyTunnelInitHandler.java

```diff
@@ -98,6 +96,31 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         initPromise.setFailure(new IOException("Could not connect to proxy"));
     }
 
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) {
+        if (!initPromise.isDone()) {
+            handleConnectRequestFailure(ctx, null);
+        }
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        if (!initPromise.isDone()) {
+            handleConnectRequestFailure(ctx, cause);
+        } else {
+            ctx.fireExceptionCaught(cause);
```

Comment:
Any reason we propagate this event, when `initPromise` is already done, but not `channelInactive`?

Reply:
Actually, there's really no need to propagate the event since this is the last handler in the pipeline. I think we should just close the channel context in this case. WDUT?

Reply:
I think in this case we're not likely to add any new handlers to this pipeline, but just to be safe let's just propagate the event if we don't handle it, which I think is the preferred default behavior.

Reply:
I think we should handle the exception though, otherwise it'll be logged as something like "unhandled exception reached the end of pipeline", which is not a good customer experience imo.

Reply:
Yeah that's what I meant; I think we agree. Let's just propagate the event if we don't call `handleConnectRequestFailure` in `channelInactive`